File: BC-W4/Army/Spells/GreaterHeal/GreaterHeal.cpp
#include "GreaterHeal.h"

GreaterHeal::GreaterHeal(int actionPoints, int manaCost, const std::string& spellName) : Spell(actionPoints, manaCost, spellName) {};
GreaterHeal::~GreaterHeal() {};

void GreaterHeal::action(Unit* enemy) {
    throw InvalidActionException();
};

void GreaterHeal::heal(Unit* unit) {
    unit->addHitPoints(this->getActionPoints());
};

File: BC-W4/ArmyJava/src/spells/GreaterHeal.java
package spells;

import exceptions.InvalidCastException;
import units.Unit;

public class GreaterHeal extends Spell {
    public GreaterHeal() {
        super("GreaterHeal", 60, 90, false);
    }

    public GreaterHeal(int actionPoints) {
        super("GreaterHeal", 60, actionPoints, false);
    }

    public void heal(Unit unit) throws InvalidCastException {
        unit.addHitPoints(this.actionPoints);
    }
}
